Update CHANGELOG for release 2.0.0

jaredledvina opened the pull request:
This PR updates the CHANGELOG for soon-to-be-released 1.3.0 version. I went through the many PR's/changes that have been merged in since 1.2.0. Nothing appears to actually have been breaking but, I'll be adding a notice in the actual release for folks to ensure they test this version prior to just deploying it. Now that we've got the issues/PR's under control w/ integration tests, these releases should be much easier and faster to cut going forward. 

Based on https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md 

@mbbroberg / @majormoses  as my first _real_ changelog update, how's this look to you?

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,35 @@
+- Initial support for OpenBSD! (@smbambling)
+- Ubuntu now get's `apt-transport-https` installed to support HTTPS repos. (@kevit)
+- Default to HTTPS APT repo's.  @jaredledvina
+- Ubuntu - Redis now always binds to `0.0.0.0` to ensure it's accessible. (@tculp)
```
I'd say that should be under a `### Breaking Changes` header and should outline the security implications of the switch. I would still question that that is a sane default over say `127.0.01`  as it is more common than you think to have servers with multiple nics and some being public. I would never do this in my environment but I have seen this done. This is not the place to debate whether or not this is a good idea, that said I think that we need to clearly warn people and ensure we version and explain it clearly. People who have multiple nics and have public interfaces will want the opportunity to spot and ensure that they choose the right interface ip.

Actually now that I think about it my sensu does have a public ip address as we want to ensure that even if our NAT instances are down alerts still come out. While we are safe by using strict aws security groups that does not mean others are as careful.

While some might argue this does not change the public api this breaks users in a very real way and my opinion semver is only helpful about informing users of risks if you actually take it beyond the letter of the law and think about how this impacts other consumers of your software.

Just to play devil's advocate: this restores a uniform experience as other platforms already had it, I could be talked into putting it under a `### Fixed` header instead.

jaredledvina replied:
So, the change in question here is, https://github.com/sensu/sensu-ansible/pull/105/files, which updated the Ubuntu config to match what we already had done with Debian: https://github.com/sensu/sensu-ansible/blob/master/tasks/Debian/redis.yml#L13-L18, CentOS: https://github.com/sensu/sensu-ansible/blob/master/tasks/CentOS/redis.yml#L19-L23, FreeBSD: https://github.com/sensu/sensu-ansible/blob/master/tasks/FreeBSD/redis.yml#L12-L17, and OpenBSD: https://github.com/sensu/sensu-ansible/blob/master/tasks/OpenBSD/redis.yml#L12-L17.

So, it does unify what we have done for the other supported operating systems. I totally agree that configuring `0.0.0.0` at all might be sketch, maybe that's something we track as an issue and possibly merge/release as a dedicated breaking change. Solely for the change in this release though, I don't feel it's a breaking change but more-so an update to ensure Ubuntu lines up with everyone else. So, maybe `Fixed` is more accurate. Although, I had considered that and I guess I wasn't sure if anything is really `Fixed` per-se but reading through https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md#fixing-a-bug, I can see how this might fit under that title. 

Now that said, while this role supports deploying the full stack, as I mentioned in  https://github.com/sensu/sensu-ansible/issues/118, ideally, for real Production use, folks don't flip the Redis/RabbitMQ bits for this role and use dedicated upstream roles for that. Mostly because this role exposes just the minimum to get both of them up in a sane way versus a dedicate role that will support everything. 

Perhaps I'm way off here, curious what you think @majormoses

majormoses replied:
Sounds good in this case to use `### Fixed` I'd still like to make sure we call out the security implications and the fact that it did change in a way that is breaking even if we choose to not call it that.

Playing devil's advocate back on a breaking change: In my env everything is Ubuntu so why do I care that debian, centos, etc were different. You changed something very important and it should not be swept under the rug.

At the end of the day I leave it to you to evaluate what you think is best but I am torn on `Fixed` or `Breaking Change` I tend to always lean more breaking change because there is no real "cost" to moving to a higher version of software and prevents users from pulling in potentially unsafe changes.

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,35 @@
-## 1.2.0 2017-05-13
+## [Unreleased]
+
+## [1.3.0] - 2018-02-04
```
I am not sure but this might be a major version, waiting on response to validate.

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,35 @@
+- Default to HTTPS APT repo's.  @jaredledvina
+- Ubuntu - Redis now always binds to `0.0.0.0` to ensure it's accessible. (@tculp)
+- Allow for configuring when a node gets the `sensu-client` config file. (@tculp)
+- Split up the variables used to determine if a host gets rabbitmq/redis for more flexibility in deployments. (@tculp)
```
Are all the old options still functional? If not this is also a breaking change.

jaredledvina replied:
Related change: https://github.com/sensu/sensu-ansible/commit/347afa18fd3a25343e6d2a17d9c5e525a811e92c

Crap, yep, version 2.0.0 here we come.

majormoses replied:
Yup, it's hard to version software...

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,35 @@
+- Amazon Linux now get's proper yum repo and supports Amazon Linux 2. (@romainrbr)
+- Yum based distros now get EPEL to support installing a newer and supported version of RabbitMQ. (@romainrbr)
+- CentOS now supports using Bintray mirrors for installing RabbitMQ to work around Erlang issues with older versions. (@romainrbr)
+- All PR's are now required to pass TravisCI tests.  (@jaredledvina)
```
:100:

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,35 @@
+- Yum based distros now get EPEL to support installing a newer and supported version of RabbitMQ. (@romainrbr)
+- CentOS now supports using Bintray mirrors for installing RabbitMQ to work around Erlang issues with older versions. (@romainrbr)
+- All PR's are now required to pass TravisCI tests.  (@jaredledvina)
+- Ensure that we configure the `mode` and `umask` for files to work in a more restrictive environment. (@roumano)
```
:100: thanks for working through the submitter's issue, I am sure they greatly appreciate it.

majormoses commented on CHANGELOG.md:
```diff
@@ -29,6 +55,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/).
 First tagged release, starting at 1.0.0 since the project can be considered stable at this point.
+
+[Unreleased]: https://github.com/sensu/sensu-ansible/compare/1.3.0...HEAD
+[1.3.0]: https://github.com/sensu/sensu-ansible/compare/1.2.0...1.3.0
```
see other comments regarding versioning.

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,41 @@
+### Fixed
+- Ubuntu - Redis now always binds to `0.0.0.0` to match other distro's configurgation and to ensure it's accessible. (@tculp)
+
+### Breaking Change
```
This should always be the first section under the version/unreleased because it is the most important for users to quickly spot and evaluate.

mbbroberg commented on CHANGELOG.md:
```diff
@@ -1,9 +1,41 @@
+### Added
+- Initial support for OpenBSD! (@smbambling)
+- Ubuntu now get's `apt-transport-https` installed to support HTTPS repos. (@kevit)
+- Default to HTTPS APT repo's.  @jaredledvina
```
s/repo's/repos/

mbbroberg commented on CHANGELOG.md:
```diff
@@ -1,9 +1,41 @@
+- Allow for deploying client definitions based on groups. (@tculp)
+- Default to HTTPS Yum repo's and install the Yum key for package signing validation via HTTPS.  (@jaredledvina)
+- Used HTTPS for APT key.  (@jaredledvina)
+- Amazon Linux now get's proper yum repo and supports Amazon Linux 2. (@romainrbr)
```
s/now get's/has/

mbbroberg commented on CHANGELOG.md:
```diff
@@ -1,9 +1,41 @@
+- Amazon Linux now get's proper yum repo and supports Amazon Linux 2. (@romainrbr)
+- Yum based distros now get EPEL to support installing a newer and supported version of RabbitMQ. (@romainrbr)
+- CentOS now supports using Bintray mirrors for installing RabbitMQ to work around Erlang issues with older versions. (@romainrbr)
+- All PR's are now required to pass TravisCI tests.  (@jaredledvina)
```
s/PR's/PRs/

mbbroberg commented on CHANGELOG.md:
```diff
@@ -1,9 +1,41 @@
+- Update 'Generate SSL Certs' to support Ansible 2.4. (@tculp)
+
+### Fixed
+- Ubuntu - Redis now always binds to `0.0.0.0` to match other distro's configurgation and to ensure it's accessible. (@tculp)
```
s/configurgation/configuration/g

jaredledvina commented:
Thanks for all the feedback, I really appreciate it! I believe I've cleaned/fixed everything.

I've also added the message for https://github.com/sensu/sensu-ansible/pull/134 which I plan on merging and including in the release. As noted in the commit message and CHANGELOG, we aren't actively breaking support for Ansible versions <2.3 however, going forward, we'll only be guaranteeing support for the last two stable versions of Ansible. This should keep everyone a bit saner :)

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,42 @@
+- Redis on Ubuntu will now be configured to bind to `0.0.0.0` to ensure accessiblity and to match the other supported OS configurations (@tculp)
+- Updated the supported Ansible version to the last two stable releases (currently that's Ansible 2.3 and 2.4) (@jaredledvina)
+  Please note that we have not explicitly broken support for running this role on versions of Ansible <2.3. However, we will only
+  be activly supporting the last two stable Ansible releases to reduce the maintenance burden.
```
why is this on a newline?

jaredledvina replied:
Was making it easier to view in Vim and I was thinking Markdown might play nice and leave it there but, I see in the rendered view that's not the case. So, nuke 'em and put it all on one line. I think the rendering looks alright.

majormoses replied:
ya when dealing with unordered lists in markdown it just is easier to put each item on its own line and let it wrap. Markdown solved that problem a long time ago.

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,42 @@
+  See the [role variable documentation](https://github.com/sensu/sensu-ansible/blob/master/docs/role_variables.md) for details on the parameters.
+- Redis on Ubuntu will now be configured to bind to `0.0.0.0` to ensure accessiblity and to match the other supported OS configurations (@tculp)
+- Updated the supported Ansible version to the last two stable releases (currently that's Ansible 2.3 and 2.4) (@jaredledvina)
+  Please note that we have not explicitly broken support for running this role on versions of Ansible <2.3. However, we will only
```
either put this on the previous line or put a `-` so it remains an item in an unordered list.

majormoses commented on CHANGELOG.md:
```diff
@@ -1,9 +1,38 @@
-## 1.2.0 2017-05-13
+## [Unreleased]
+
+## [2.0.0] - 2018-02-07
```
Generally you don't version it in the PR but this is a special case so it's fine.
